### PR TITLE
Add sub_user support to authenticated calls

### PR DESCRIPTION
**Pull request by adeadman**
`get_auth_params()` now checks if a sub user has been specified, and if so adds it to the parameters when making a request.

Updated tests to match

### CODE REVIEW

File: features/steps/step_reservation.py [hunk 1]
```diff
@@ -253,18 +253,11 @@ def then_my_trolley_does_not_contain_tickets_for_event(context, event_id):
 @then('my trolley contains the requested seats')
 def then_my_trolley_contains_the_requested_seats(context):
     order = context.trolley.bundles[0].orders[0]
-    seat_ids = [seat.id for seat in order.requested_seats]
+    seat_ids = order.requested_seat_ids #[seat.id for seat in order.requested_seats]
```
Comment (nicwest) [inline features/steps/step_reservation.py]: legacy code

File: pyticketswitch/client.py [hunk 2]
```diff
@@ -87,7 +87,11 @@ def get_auth_params(self):
             dict: auth params that passed to requests
 
         """
-        return {'user_id': self.user, 'user_passwd': self.password}
+        if self.sub_user:
+            return {'user_id': self.user, 'user_passwd': self.password,
+                    'sub_user': self.sub_user}
+        else:
```
Comment (nicwest) [inline pyticketswitch/client.py]: don' need this

File: features/steps/step_reservation.py (hunk 1 above)
Comment (adeadman) [inline features/steps/step_reservation.py]: Fixed

File: pyticketswitch/client.py (hunk 2 above)
Comment (adeadman) [inline pyticketswitch/client.py]: Gone

**Pull request merged**